src/components/Aside.astro

```diff
@@ -0,0 +1,126 @@
+---
+import UIString from './UIString.astro';
+export interface Props {
+	type?: 'note' | 'tip' | 'info' | 'caution' | 'danger' | 'learn';
+	title?: string;
+}
+
+const { type = 'note', title } = Astro.props as Props;
+const id = 'aside-' + (Math.random() * 10e16).toString(32);
```

Comment:
I'm not particularly happy with the idea of having a random output in an Astro component. If it's only used for `aria-labelledby`, can you instead change that to `aria-label={title}` and thereby get rid of the need for an ID completely?

Reply:
I think the issue is that we then would need to _hide_ the actual text node with the title from screenreaders if we did this or the title would be repeated, so I’d prefer this pattern if possible. What’s your concern with randomising the IDs? That there’s a small chance of conflicts? Would a more robust ID library like `nanoid` reassure you?

Reply:
No, my point is that I like to have stable build output in general.

I often do full directory comparisons of the build output to catch if an upgrade of some dependency introduced changes to the HTML output. This becomes harder with each piece of randomness added to the output.

I honestly wonder if there is any actual need for an aria label in the current case. The contained text should be perfectly accessible by default, don't you think?

Reply:
I believe the labelling (which I [based on the BBC’s GEL markup](https://bbc.github.io/gel/components/breakout-boxes/#recommended-markup)) is motivated by this: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#labeling_section_content

A user might be offered to hear a list of sections and labelling gives the section a title.

How about I hash the title and contents of the Aside?

Reply:
Hmm, I see. Hashing wouldn't work I'm afraid, because the exact same Aside content could be repeated multiple times on a page.

What about my idea of using `aria-label={title}` on the container and then just adding `aria-hidden="true"` to the title <p> to solve your concern of it being read out twice?

Reply:
Seems to work well with VoiceOver on Mac — and the Accessibility Tree in Firefox dev tools also looks OK. Would love some more tests. @Yan-Thomas Want to try this out with your screenreader?

Reply:
I can test this out tomorrow (I'm on my sister's laptop so I don't have my screen reader here, unfortunately). Sorry for the late response, this mention didn't notify me for some reason?! - I'm not even sure if you still need to test this out :sweat_smile:

Reply:
Yes, if you could, a test would be helpful! I could only test on macOS VoiceOver on my end and the more tests the better 💜